backend/src/routes/v0/server.py

```
@@ -1,154 +1,158 @@
-class Config(BaseModel):
-    type: str = Field(
-        ..., description="Server type: 'mcp' or 'a2a'", pattern="^(mcp|a2a)$"
-    )
-    config: dict
-
+def _resolve_and_validate_host(url: str) -> None:
+    """Resolve DNS for a URL and check the resolved IP is not private (DNS rebinding protection)."""
+    parsed = urlparse(url)
+    hostname = parsed.hostname
+    if not hostname:
+        raise ValueError("Invalid URL: no hostname")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)